retryrpc/server.go

```diff
@@ -405,13 +433,16 @@ func (server *Server) trimCompleted(t time.Time, long bool) {
 		// lock.
 		for e := l.Front(); e != nil; e = e.Next() {
 			key := e.Value.(string)
-			v := server.perClientInfo[key]
+			ci := server.perClientInfo[key]
 
-			v.Lock()
-			if v.isEmpty() {
+			ci.Lock()
+			ci.cCtx.Lock()
+			if ci.isEmpty() && ci.cCtx.serviceClientExited == true {
```

Comment:
Yummy! Love these one line changes for some challenging issues!!!

Reply:
yes - very isolated
I have not added bucketstats yet. I am trying to do that today but wanted to get this merged